warehouse/oidc/models/_core.py

```diff
@@ -23,8 +23,14 @@
 from warehouse.oidc.interfaces import SignedClaims
 from warehouse.packaging.models import Project
 
+C = TypeVar("C")
 
-def _check_claim_binary(binary_func):
+CheckClaimCallable = Callable[[C, C, Any], bool]
```

Comment:
Nit: you can ratchet down the `Any` here to `SignedClaims` 🙂 

(These claim checking functions should only ever take a `SignedClaims` in that position.)

Reply:
This was a type that I had thought about adding but I had over complicated in my head now that I see `SignedClaims`.  Thanks for pointing it out!